File: API/ICCMS-API/Models/Phase.cs
using Google.Cloud.Firestore;

namespace ICCMS_API.Models
{
    [FirestoreData]
    public class Phase
    {
        [FirestoreProperty("phaseId")]
        public string PhaseId { get; set; } = string.Empty;

        [FirestoreProperty("projectId")]
        public string ProjectId { get; set; } = string.Empty;

        [FirestoreProperty("name")]
        public string Name { get; set; } = string.Empty;

        [FirestoreProperty("description")]
        public string Description { get; set; } = string.Empty;

        [FirestoreProperty("startDate")]
        public DateTime StartDate { get; set; }

        [FirestoreProperty("endDate")]
        public DateTime EndDate { get; set; }

        [FirestoreProperty("status")]
        public string Status { get; set; } = string.Empty;

        [FirestoreProperty("progress")]
        public int Progress { get; set; } = 0;

        [FirestoreProperty("budget")]
        public double Budget { get; set; }

        [FirestoreProperty("assignedTo")]
        public string AssignedTo { get; set; } = string.Empty;
    }
}

File: Mobile/app/src/main/java/com/example/iccms_mobile/data/models/Phase.kt
package com.example.iccms_mobile.data.models

import com.google.gson.annotations.SerializedName

data class Phase(
    @SerializedName("phaseId")
    val phaseId: String = "",

    @SerializedName("projectId")
    val projectId: String = "",

    @SerializedName("name")
    val name: String = "",

    @SerializedName("description")
    val description: String = "",

    @SerializedName("startDate")
    val startDate: String = "",

    @SerializedName("endDate")
    val endDate: String = "",

    @SerializedName("status")
    val status: String = "",

    @SerializedName("progress")
    val progress: Int = 0,

    @SerializedName("budget")
    val budget: Double = 0.0,

    @SerializedName("assignedTo")
    val assignedTo: String = ""
)
